gallery/src/components/NxFontAwesomeIcon/NxFontAwesomeIconExample.tsx

```diff
@@ -27,6 +27,12 @@ const NxFontAwesomeIconExample = () =>
         */}
       <NxFontAwesomeIcon icon={faBatteryEmpty} color="red" />
     </button>
+    {/* Icon only variant button with aria-label on the icon, note that aria-hidden has been set to false */}
+    <NxButton variant="icon-only">
+      <NxFontAwesomeIcon icon={faEdit} aria-hidden="false" aria-label="Edit text" />
```

Comment:
We have icon-only button examples in the `NxButton` page which are in fact written differently than this - they put the `aria-label` directly on the button.  We don't want a conflicting example here.

Reply:
> We have icon-only button examples in the `NxButton` page which are in fact written differently than this - they put the `aria-label` directly on the button. We don't want a conflicting example here.

I thought about this a bunch and IMO `aria-label` should be applied to the icon because that is what it is labelling, not the button. I had avoided modifying the NxButton examples (for now) because i didn't want to overlap with anybody else's work but I think that they should be modified to follow this pattern. I'm happy to take that on as a part of this ticket.

Reply:
> I thought about this a bunch and IMO aria-label should be applied to the icon because that is what it is labelling, not the button. I had avoided modifying the NxButton examples (for now) because i didn't want to overlap with anybody else's work but I think that they should be modified to follow this pattern. I'm happy to take that on as a part of this ticket.

TBH I have the opposite philosophy about it.  I think the semantics are most accurately associated with the button, and the icon is a presentational concern within the button.  Think of it this way: you may have two buttons in different places which use the same icon to mean different things, especially if you craft that meaning in a more specific way (For instance, "delete waiver" rather than just "delete").  It's not that there's an icon that inherently means "delete waiver", it's that there's a particular button on that particular page that has that meaning.

The fact that doing it that way avoids the need for the `aria-hidden` override is an added bonus.

Another, bigger thought: should icon-only buttons always have tooltips? Should we ideally make a separate `NxIconOnlyButton` component that renders an `NxButton` wrapped in an `NxTooltip`?  That would be an alternative way to label the buttons and avoid the question above.

Reply:
> > I thought about this a bunch and IMO aria-label should be applied to the icon because that is what it is labelling, not the button. I had avoided modifying the NxButton examples (for now) because i didn't want to overlap with anybody else's work but I think that they should be modified to follow this pattern. I'm happy to take that on as a part of this ticket.
> 
> TBH I have the opposite philosophy about it. I think the semantics are most accurately associated with the button, and the icon is a presentational concern within the button. Think of it this way: you may have two buttons in different places which use the same icon to mean different things, especially if you craft that meaning in a more specific way (For instance, "delete waiver" rather than just "delete"). It's not that there's an icon that inherently means "delete waiver", it's that there's a particular button on that particular page that has that meaning.
> 
> The fact that doing it that way avoids the need for the `aria-hidden` override is an added bonus.
> 
> Another, bigger thought: should icon-only buttons always have tooltips? Should we ideally make a separate `NxIconOnlyButton` component that renders an `NxButton` wrapped in an `NxTooltip`? That would be an alternative way to label the buttons and avoid the question above.

I think that your point is valid, however I would prefer to do that work in a  separate `NxButton` PR.

Reply:
> I would prefer to do that work in a separate NxButton PR.

I assume you are referring here to the very last thing I said?  Everything I said describes the way the NxButton examples already are.

Reply:
I have created https://issues.sonatype.org/browse/RSC-398 for the button work.